### PR TITLE
Allow more granularly setting nested dependencies

### LINKED ISSUE #159: Allow more granularly setting nested dependencies
Right now I have a setup where I publish 2 variants of jars:
- One without dependencies (modpack)
- One with dependencies (standalone)

Now I would like to extract a module out of my mod to be used in another one. There's not much point distributing the module because it will likely only be used by me. So what I would like to do is include it in _both_ variants. However, if I set `addNestedDependencies` to `true`, all dependencies will be added, and if I set it to false, none of them will be added. I need a way to include _only_ the one shared module. 
I think the best way of going about these is having a `forceAddNestedDependencies` (or a better name if you can think of one) on the `RemapJarTask` that is a list of `archivesBaseNames` that will be included even when `addNestedDependencies` are true.

### PR DESCRIPTION
**Pull request by natanfudge**
Closes #159

### CODE REVIEW

**Comment by natanfudge** [comment]
On second thought, wouldn't it make more sense to have `include` as a concept be tied to the `Jar` task, instead of something you apply to everything? This way you just choose exactly what dependencies you want to include for each `Jar` task.

File: src/main/java/net/fabricmc/loom/task/RemapJarTask.java [hunk 1]
```diff
@@ -122,10 +126,9 @@ public void doTask() throws Throwable {
 			project.getLogger().debug("Transformed mixin reference maps in output JAR!");
 		}
 
-		if (getAddNestedDependencies().getOrElse(false)) {
-			if (NestedJars.addNestedJars(project, output)) {
-				project.getLogger().debug("Added nested jar paths to mod json");
-			}
+		if (NestedJars.addNestedJars(project, output, getForcedNestedDependencies().getOrElse(new ArrayList<>()),
```
Comment (liach) [inline src/main/java/net/fabricmc/loom/task/RemapJarTask.java]: use collections emptylist call

File: src/main/java/net/fabricmc/loom/task/RemapJarTask.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.ArrayList;
```
Comment (liach) [inline src/main/java/net/fabricmc/loom/task/RemapJarTask.java]: Remove this import

File: src/main/java/net/fabricmc/loom/task/RemapJarTask.java [hunk 3]
```diff
@@ -28,11 +28,14 @@
 import java.io.FileNotFoundException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
```
Comment (natanfudge) [inline src/main/java/net/fabricmc/loom/task/RemapJarTask.java]:
```suggestion
```

**Review by liach — APPROVED** [review]

**Comment by natanfudge** [comment]
Probably not relevant if you use a modpack flattener

**Pull request closed without merging**